server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/WriteLoadConstraintDecider.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.cluster.routing.allocation.decider;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.routing.RoutingNode;
+import org.elasticsearch.cluster.routing.ShardRouting;
+import org.elasticsearch.cluster.routing.allocation.RoutingAllocation;
+import org.elasticsearch.cluster.routing.allocation.WriteLoadConstraintSettings;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.threadpool.ThreadPool;
+
+/**
+ * Decides whether shards can be allocated to cluster nodes, or can remain on cluster nodes, based on the target node's current write thread
+ * pool usage stats and any candidate shard's write load estimate.
+ */
+public class WriteLoadConstraintDecider extends AllocationDecider {
+    private static final Logger logger = LogManager.getLogger(WriteLoadConstraintDecider.class);
+
+    public static final String NAME = "write_load";
+
+    private final WriteLoadConstraintSettings writeLoadConstraintSettings;
+
+    public WriteLoadConstraintDecider(ClusterSettings clusterSettings) {
+        this.writeLoadConstraintSettings = new WriteLoadConstraintSettings(clusterSettings);
+    }
+
+    @Override
+    public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, RoutingAllocation allocation) {
+        if (writeLoadConstraintSettings.getWriteLoadConstraintEnabled() != WriteLoadConstraintSettings.WriteLoadDeciderStatus.ENABLED) {
+            return Decision.YES;
+        }
```

Comment:
I had made the `WriteLoadDeciderStatus` have two levels (`LOW_ONLY` and `ENABLED`) the former being for `canAllocate` and the latter being for `canRemain`.

I don't know if we still want to enable these incrementally like that but if so, I think we should only return `YES` here if the setting is `DISABLED` (i.e. if it's either `LOW_ONLY` or `ENABLED`)

Reply:
Fixed 👍